hax/hax/motr/hax.c

```diff
@@ -594,39 +605,32 @@ static void link_absent_cb(struct m0_halon_interface *hi,
 }
 
 static void link_is_disconnecting_cb(struct m0_halon_interface *hi,
-				     struct m0_ha_link *hl)
+				     struct m0_ha_link *link)
 {
-	M0_LOG(M0_DEBUG, "disconnecting ha link to %s",
-		m0_rpc_conn_addr(&hl->hln_rpc_link.rlk_conn));
-	m0_halon_interface_disconnect(hi, hl);
+	struct hax_link *hxl;
+
+	hax_lock(hc0);
+	hxl = m0_tl_find(hx_links, l, &hc0->hc_links, l->hxl_link == link);
+	if (hxl != NULL)
```

Comment:
[* defect *] Here and elsewhere: if there is no hlx in the list then it's a bug. Please don't ignore logic errors.

Reply:
So what do you suggest? M0_ASSERT()?

Reply:
Yes. There is no way to continue meaningful work in case if there is a logic error here.

Reply:
Added M0_ASSERT.

Reply:
Please do it in the rest of the places when it could be a bug like this as well.